src/module.rs

```diff
@@ -30,41 +30,96 @@ impl WasmModule {
         }
     }
 
+    pub fn imports(&self) -> impl Iterator<Item = &ImportEntry> {
+        self.module
+            .import_section()
+            .map_or(Either::Left(iter::empty()),
+                    |section| Either::Right(section.entries().iter()))
+    }
+
+    /// Safe function for counting the number of imported functions.
+    /// Use this instead of `self.imports().size_hint()`!
+    pub fn count_imported_functions(&self) -> usize {
```

Comment:
I think this is a good place to have an abstraction boundary, especially since the `module` field is private so we wouldn't be able to access parity's functions from other (Rust) modules.